frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+import { t } from "ttag";
+import { useSelector } from "metabase/lib/redux";
+import { getApplicationName } from "metabase/selectors/whitelabel";
+import Modal from "metabase/components/Modal";
+import { Box } from "metabase/ui";
+import * as MetabaseAnalytics from "metabase/lib/analytics";
+import { ModalContentActionIcon } from "metabase/components/ModalContent";
+import { color } from "metabase/lib/colors";
+
+import type { EmbedType } from "./EmbedModal.types";
+import { EmbedModalHeader } from "./EmbedModal.styled";
+
+interface EmbedModalProps {
+  isOpen?: boolean;
+  onClose: () => void;
+  children: ({
+    embedType,
+    setEmbedType,
+  }: {
+    embedType: EmbedType;
+    setEmbedType: (type: EmbedType) => void;
+  }) => JSX.Element;
+}
+
+export const EmbedModal = ({ children, isOpen, onClose }: EmbedModalProps) => {
+  const [embedType, setEmbedType] = useState<EmbedType>(null);
+  const applicationName = useSelector(getApplicationName);
+
+  const isEmbeddingStage = embedType === "application";
+
+  const onEmbedClose = () => {
+    MetabaseAnalytics.trackStructEvent("Sharing Modal", "Modal Closed");
+    onClose();
+    setEmbedType(null);
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={onEmbedClose}
+      title={!isEmbeddingStage ? t`Embed ${applicationName}` : undefined}
+      fit
+      formModal={false}
+    >
+      <Box bg={isEmbeddingStage ? color("white") : "bg.0"} h="100%">
```

Comment:
I wouldn't want to juggle styles and markups to accommodate some differences in the design. I'd rather stay more simpler.

Here's my attempt to simplify this

```ts
    <Modal
      isOpen={isOpen}
      onClose={onEmbedClose}
      title={
        !isEmbeddingStage ? (
          t`Embed ${applicationName}`
        ) : (
          <>
            <ModalContentActionIcon
              name="chevronleft"
              onClick={() => setEmbedType(null)}
            />
            {t`Static embedding`}
          </>
        )
      }
      fit
      formModal={false}
    >
      <Divider />
      {children({ embedType, setEmbedType })}
    </Modal>
```
| My version                                                                                           | This PR                                                                                          |
|----------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------|
| ![image](https://github.com/metabase/metabase/assets/1937582/b5a6c4e7-9b6b-4081-a7a0-f5494dff70d9) | ![image](https://github.com/metabase/metabase/assets/1937582/e935ac6a-36ea-4498-86b2-df40adcb7c07) |


The visual difference is minimal, but we can remove some markup complexity.

This makes sure the `title` prop takes care of both title situation, so it should be easier to reason about. 
it also uses Mantine's `Divider` which is a very good component, so we don't need to create another styled component just to have a border, we can also remove `EmbedModalHeader` and its file because of that. And lastly, `children` prop doesn't have different parent containers (or parents with different styles which is a bit hard to read since it had a conditional style)